search/engine.go

```diff
@@ -762,6 +788,49 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 		resultsByLang[lang] = append(resultsByLang[lang], tweets)
 	}
 
+	filteredByLang := <-grammarsFilteredResultsByLangChannel
```

Comment:
Comment what this loop does... Consider extracting those loops to functions with documentation (same as the comment before the loop).

Reply:
comment added